alica_engine/src/engine/RunningPlan.cpp

```diff
@@ -584,22 +583,18 @@ void RunningPlan::activate()
         _basicPlan = static_cast<const Plan*>(_activeTriple.abstractPlan)->getBasicPlan();
     }
 
-    auto& scheduler = _ae->editScheduler();
-    int jobID = scheduler.getNextJobID();
-    std::function<void()> cb;
-
     if (_basicPlan) {
-        cb = std::bind(&BasicPlan::init, _basicPlan);
-    }
-
-    std::vector<std::weak_ptr<scheduler::Job>> prerequisites = getDeactivatedSiblingsTerminateJobs();
-    if (_parent) {
-        prerequisites.emplace_back(_parent->getInitJob());
+        auto& scheduler = _ae->editScheduler();
```

Comment:
better to move this code to BasicPlan & BasicBehaviour respectively. Same thing for termination. There are already startBehaviour & stopBehaviour methods, similarly we can add such methods to BasicPlan & call them here instead of runningPlan having to create & schedule these jobs.